app/app.go

```diff
@@ -1333,6 +1337,29 @@ func (app *App) GetName() string {
 	return app.Name
 }
 
+// GetUUID returns the app v4 UUID. An UUID will be generated
+// if it does not exist.
+func (app *App) GetUUID() (string, error) {
+	if app.UUID != "" {
+		return app.UUID, nil
+	}
+	uuidV4, err := uuid.NewV4()
+	if err != nil {
+		return "", errors.WithMessage(err, "failed to generate uuid v4")
+	}
+	app.UUID = uuidV4.String()
```

Comment:
By setting `app.UUID` here future calls to `GetUUID()` will always return it even if an error occurred while storing it on the database and this value won't ever be stored. It would be safer to only set `app.UUID` after a successful call to DB Update.

Reply:
nice catch. updated